apluslms_roman/cli.py

```diff
@@ -305,13 +326,23 @@ def build_action(context):
     config = get_config(context)
     engine = get_engine(context)
 
+    builder = engine.create_builder(config)
+
+    # TODO: option to print more?
+    if (context.parser.parse_args().steps):
+        steps = builder.get_steps()
+        num_len = len(str(len(steps)))
+        name_len = len(max(steps, key=lambda s: len(s.name or "")).name)
+        for step in steps:
+            step_str = "{:%dd}. {:%ds} {}" % (num_len, name_len)
+            print(step_str.format(step.ref, step.name or "", step.img))
+        return
+
     if not verify_engine(engine, only_when_error=True):
         return 1
 
     # build course
-    builder = engine.create_builder(config)
-    result = builder.build()
-    print(result)
+    result = builder.build(context.parser.parse_args().buildsteps)
```

Comment:
list of steps could be prepared using code `steps = list(chain.from_iterable(x.split(',') for x in context.args.steps))` (chain is from `itertools`). This way user can also say `--steps=foo,bar` instead of just `-s foo -s bar`, but both would be supported.

Reply:
`-s foo -s bar` doesn't work, it only builds bar. `-s foo bar` or `--steps foo bar` work, do we still want to support `--steps=foo,bar` too?

Reply:
First time I actually where thinking that `--steps` used `action='append'` instead of `nargs='+'`, thus I thought `-s foo -s bar` would be the syntax. My bad. Any case, more comments on the topic on the new version of the file.

Reply:
chain doesn't work, used an if clause instead